interface/reports/inventory_list.php

```diff
@@ -26,46 +26,662 @@
     if ($warnings) {
-        $warnings .= '<br />';
+        $warnings .= $break;
     }
-
     $warnings .= text($msg);
 }
 
+// Check if a product needs to be re-ordered, optionally for a given warehouse.
+//
+function checkReorder($drug_id, $min, $warehouse = '')
+{
+    global $form_days;
+
+    if (!$min) {
+        return false;
+    }
+
+    $query = "SELECT " .
+        "SUM(s.quantity) AS sale_quantity " .
+        "FROM drug_sales AS s " .
+        "LEFT JOIN drug_inventory AS di ON di.inventory_id = s.inventory_id " .
+        "WHERE " .
+        "s.drug_id = ? AND " .
+        "s.sale_date > DATE_SUB(NOW(), INTERVAL ? DAY) " .
+        "AND s.pid != 0";
+    $binds = array($drug_id, $form_days);
+    if ($warehouse !== '') {
+        $query .= " AND di.warehouse_id = ?";
+        $binds[] = $warehouse;
+    }
+    $srow = sqlQuery($query, $binds);
+    $sales = 0 + $srow['sale_quantity'];
+
+    $query = "SELECT SUM(on_hand) AS on_hand " .
+      "FROM drug_inventory AS di WHERE " .
+      "di.drug_id = ? AND " .
+      "(di.expiration IS NULL OR di.expiration > NOW()) AND " .
+      "di.destroy_date IS NULL";
+    $binds = array($drug_id);
+    if ($warehouse !== '') {
+        $query .= " AND di.warehouse_id = ?";
+        $binds[] = $warehouse;
+    }
+    $ohrow = sqlQuery($query, $binds);
+    $onhand = intval($ohrow['on_hand']);
+
+    if (empty($GLOBALS['gbl_min_max_months'])) {
+        if ($onhand <= $min) {
+            return true;
+        }
+    } else {
+        if ($sales != 0) {
+            $stock_months = sprintf('%0.1f', $onhand * ($form_days / 30.41) / $sales);
+            if ($stock_months <= $min) {
+                return true;
+            }
+        }
+    }
+
+    return false;
+}
+
+// Generate the list of warehouse IDs that the current user is allowed to access.
+// This is used to build SQL for $uwcond.
+//
+function genUserWarehouses($userid = 0)
+{
+    $list = '';
+    $res = sqlStatement(
+        "SELECT DISTINCT option_id, option_value FROM list_options WHERE " .
+        "list_id = 'warehouse' AND activity = 1"
+    );
+    while ($row = sqlFetchArray($res)) {
+        if (isWarehouseAllowed($row['option_value'], $row['option_id'], $userid)) {
+            if ($list != '') {
+                $list .= ', ';
+            }
+            $list .= "'" . $row['option_id'] . "'";
+        }
+    }
+    return $list;
+}
+
+// This counts the number of days that have a starting zero inventory for a given product
+// since a given start date with given restrictions for warehouse or facility.
+// End date is assumed to be the current date.
+//
+// function zeroDays($product_id, $begdate, $warehouse_id = '~', $facility_id = 0) {
+function zeroDays($product_id, $begdate, $extracond, $min_sale = 1)
+{
+    $today = date('Y-m-d');
+
+    $prodcond = '';
+    if ($product_id) {
+        $prodcond = "AND di.drug_id = '" . add_escape_custom($product_id) . "'";
```

Comment:
better to use sql binding, if possible

Reply:
Because $extracond is passed in that would require some major code changes. What I'll do is change these queries to use manual escaping only. That should work, no?

Reply:
It would help if binding were more flexible, as in the last example here:
http://adodb.org/dokuwiki/doku.php?id=v5:reference:connection:execute

... however I tested that and it does not work.

Reply:
Never mind, I figured out a way to deal with this without making a mess.